.github/workflows/build.yml

```diff
@@ -31,36 +31,36 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository.
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Setup NodeJS
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
-          node-version: '16'
+          node-version: '18'
       - name: Install modules.
         run: yarn
       - name: Build artifacts.
         run: yarn dist:linux appimage --publish=never
       - name: Upload AppImage.
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
           name: linux-appimage
           path: dist/Heroic-*.AppImage
           retention-days: 14
   build_mac:
-    runs-on: macos-11
+    runs-on: macos-latest
     steps:
       - name: Checkout repository.
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Setup NodeJS
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
-          node-version: '16'
+          node-version: '18'
       - name: Install modules.
         run: yarn
       - name: Build artifacts.
         run: yarn dist:mac --publish=never
       - name: Upload DMG.
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
```

Comment:
just one question cause I don't know how this works, I see this `Upload DMG.` step that was not split in two like the one in the `build-prs-mac.yml` file that now has `Upload x64.` and `Upload ARM64.` steps.

Should this one be split too?

Reply:
good point